### DISCUSSION TITLE
entity for battery min soc / battery capacity

### DISCUSSION THREAD

**Discussion by vip-klaus** [nathanmarlor/foxess_modbus#444] [Support]
Hello everyone,

I use an H3-12.0-E inverter and an ECS2900-H7 battery and am currently trying to create a template entity in HA that calculates how long the battery will last at the current load. To do this, I need the entity 'foxess_inv1_min_soc_on_grid' as well as the information about how many battery cells are installed or how large the total battery capacity is.
Are these entities made available in the FoxESS-Madbus integration? Unfortunately I can't find it in HA. There is obviously only the 'inv_battery_soc' entity that describes the current charge state.

Can someone answer that?

Thanks and Regards
Klaus

**Comment by canton7** [comment]
Search the entities list for "SoC" and you'll find `min_soc_on_grid`. We don't currently expose the number of cells or total capacity.

**Reply by vip-klaus** [reply]
@canton7 thank you for your fast reply.

Of course I searched for entities with SoC, but didn't find any for the min SoC.

I think I've found the reason for this: in entity_descriptions.py this entity is not made available for the H3 inverter. Only for the H1, AIO, AC1 and KH (see appendix) Doesn't address 41011 apply to the H3? Could I just try to include the H3 in the description, or could that destroy my inverter?

Thanks and regards
Klaus

![min_soc](https://github.com/nathanmarlor/foxess_modbus/assets/144471901/16b7cb41-6b03-4c18-89d9-f394c42bef03)

**Reply by canton7** [reply]
Aah sorry you've got an H3. As far as we know, such registers aren't exposed by the H3 unfortunately.

**Reply by canton7** [reply]
(The H3 doesn't expose any input registers at all, only holding registers)

**Reply by vip-klaus** [reply]
I suspected that. But how does an app like Energy-Stats know the minimum SoC? Will this be communicated via the cloud?
